tests/scripts/cmd_utils.py

```diff
@@ -44,18 +44,23 @@ def init_log():
 
 
 class Sh:
-    def __init__(self, env=None):
+    def __init__(self, env=None, cwd=None):
         self.env = os.environ.copy()
         if env is not None:
             self.env.update(env)
+        self.cwd = cwd
 
     def run(self, cmd: str, **kwargs):
         logging.info(f"+ {cmd}")
-        if "check" not in kwargs:
-            kwargs["check"] = True
-        if "shell" not in kwargs:
-            kwargs["shell"] = True
-        if "env" not in kwargs:
-            kwargs["env"] = self.env
-
-        subprocess.run(cmd, **kwargs)
+        defaults = {
+            "check": True,
+            "shell": True,
+            "env": self.env,
+            "encoding": "utf-8",
+            "cwd": self.cwd,
+        }
+        for k, v in defaults.items():
+            if k not in kwargs:
+                kwargs[k] = v
```

Comment:
Minor minor nit, would be nicer not to mutate `kwargs` and instead produce a new dictionary here?

```
subprocess_args = {
  k: v if k not in kwargs else kwargs[k]
  for k, v in defaults.items()
}
```